public/docs/ts/latest/guide/component-communication.jade

```diff
@@ -0,0 +1,707 @@
+
++makeExample('component-communication/ts/app/invite-heroes/hero-panel.ts', 'input-def', null, {pnk: /(@Input\(\))/g})
+
+:marked
+.alert.is-critical
+  :marked
+    It's easy to forget the parentheses. Our application will fail mysteriously if we do.
+:marked
+  Instead of using the `@Input()` annotation, we can use the `inputs` annotation property of `@Component()` to specify input properties.
+
+  Using the alternative method, we could have defined the `hero` input property this way:
+  
++makeExample('component-communication/ts/fragments/using-inputs.ts', null, null, {pnk: /(inputs)/g})
+
+:marked
+.alert.is-helpful
+  :marked
+    In this chapter, we are age going to use the `@Input()` annotation.
+:marked
+  ### The convenience of input data binding
+  
+  The change detection mechanism of Angular makes input data binding easy-to-use. We can utilize the input property values in expressions, too,
+  and it makes displaying data simple.
+  
+  We have already implemented the functionality behind the __Invite heroes__ function with input data binding. The same mechanism allows us
+  to implement announcing jobs. As the altered definition of `HeroPanel` shows, now it has another input property, `request`, which has two
+  occurrences within the template:
+
++makeExample('component-communication/ts/app/send-job-request/hero-panel.ts', 'component', 'send-job-request/hero-panel.ts (extract)', {pnk: /(&quot;request&quot;|{{request \|\|.*\}\})/g})
+
+:marked
+  As the value of `request` changes, the value of the `class` attribute is updated, as well as the text of the `<h4>` tag through the 
+  `request || 'No job announced'` expression. The `HeroJobBoard` side of the input data binding is as simple as we expect:
+  
++makeExample('component-communication/ts/fragments/sending-request.ts', null, 'send-job-request/hero-job-board.ts (extract)', {pnk: /(\[request\]=request)/g})
+
+:marked
+  After inviting heroes, clicking the **Ask** button invokes `announceJob()` that sets the `request` property of `HeroJobBoard` &mdash; and Angular propagates this
+  change to all `HeroPanel` instances:
+
+figure.image-display
+  img(src="/resources/images/devguide/component-communication/send-job-request-ui.png" alt="Job request is sent")
+  
+:marked
+  Now, let's see how a child can communicate with its parent.
+  
+  ## #2: Child to parent communication with event emitting
+  
+  Input property binding is an easy-to-use technique that works great in most parent-to-child communication scenarios. As the expression bound to
+  the input property changes, the modification is immediately propagated to all children. At first we may think that this mechanism could be
+  applied in the opposite direction, to send data from a child to its parent.
+  
+  _Angular does not implement input data binding in this direction._
+  
+  It is a deliberate design decision. Angular was designed with _unidirectional data flow_ in mind. This approach allows better performance 
```

Comment:
It's good point you make about no reverse property binding from child to parent. Enough to observe that all data flows are parent to child during a single JS turn. Flows from child to parent must occur in a subsequent JS turn. This is an aspect of unidirectional data flow (I call it uni-flow for short ;-))

Let's skip the discussion of why? Just cut these sentences and move on.